File: xtream/src/main/java/com/m3u/data/database/model/Channel.java
package com.m3u.data.database.model;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "channels")
public class Channel {

    @ColumnInfo(name = "url")
    public String url;

    @ColumnInfo(name = "group")
    public String category;

    @ColumnInfo(name = "title")
    public String title;

    @ColumnInfo(name = "cover")
    public String cover;

    @ColumnInfo(name = "playlistUrl", index = true)
    public String playlistUrl;

    @ColumnInfo(name = "license_type", defaultValue = "NULL")
    public String licenseType;

    @ColumnInfo(name = "license_key", defaultValue = "NULL")
    public String licenseKey;

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    public int id;

    @ColumnInfo(name = "favourite", index = true)
    public boolean favourite;

    @ColumnInfo(name = "hidden", defaultValue = "0")
    public boolean hidden;

    @ColumnInfo(name = "seen", defaultValue = "0")
    public long seen;

    @ColumnInfo(name = "channel_id", defaultValue = "NULL")
    public String originalId;

    // Static constants for license types
    public static final String LICENSE_TYPE_WIDEVINE = "com.widevine.alpha";
    public static final String LICENSE_TYPE_CLEAR_KEY = "clearkey";
    public static final String LICENSE_TYPE_CLEAR_KEY_2 = "org.w3.clearkey";
    public static final String LICENSE_TYPE_PLAY_READY = "com.microsoft.playready";

    // Constructor
    public Channel(String url, String category, String title, String cover, String playlistUrl,
                   String licenseType, String licenseKey, int id, boolean favourite, boolean hidden,
                   long seen, String originalId) {
        this.url = url;
        this.category = category;
        this.title = title;
        this.cover = cover;
        this.playlistUrl = playlistUrl;
        this.licenseType = licenseType;
        this.licenseKey = licenseKey;
        this.id = id;
        this.favourite = favourite;
        this.hidden = hidden;
        this.seen = seen;
        this.originalId = originalId;
    }

    public Channel(
            String url,
            String category,
            String title,
            String cover,
            String playlistUrl,
            String originalId
    ) {
        this.url = url;
        this.category = category;
        this.title = title;
        this.cover = cover;
        this.playlistUrl = playlistUrl;
        this.originalId = originalId;
    }

}

File: xtream/src/main/java/com/m3u/data/database/model/Channel.kt
package com.m3u.data.database.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "channels")
class Channel {
    @ColumnInfo(name = "url")
    var url: String?

    @ColumnInfo(name = "group")
    var category: String?

    @ColumnInfo(name = "title")
    var title: String?

    @ColumnInfo(name = "cover")
    var cover: String?

    @ColumnInfo(name = "playlistUrl", index = true)
    var playlistUrl: String?

    @ColumnInfo(name = "license_type", defaultValue = "NULL")
    var licenseType: String? = null

    @ColumnInfo(name = "license_key", defaultValue = "NULL")
    var licenseKey: String? = null

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id: Int = 0

    @ColumnInfo(name = "favourite", index = true)
    var favourite: Boolean = false

    @ColumnInfo(name = "hidden", defaultValue = "0")
    var hidden: Boolean = false

    @ColumnInfo(name = "seen", defaultValue = "0")
    var seen: Long = 0

    @ColumnInfo(name = "channel_id", defaultValue = "NULL")
    var originalId: String?

    // Constructor
    constructor(
        url: String?, category: String?, title: String?, cover: String?, playlistUrl: String?,
        licenseType: String?, licenseKey: String?, id: Int, favourite: Boolean, hidden: Boolean,
        seen: Long, originalId: String?
    ) {
        this.url = url
        this.category = category
        this.title = title
        this.cover = cover
        this.playlistUrl = playlistUrl
        this.licenseType = licenseType
        this.licenseKey = licenseKey
        this.id = id
        this.favourite = favourite
        this.hidden = hidden
        this.seen = seen
        this.originalId = originalId
    }

    constructor(
        url: String?,
        category: String?,
        title: String?,
        cover: String?,
        playlistUrl: String?,
        originalId: String?
    ) {
        this.url = url
        this.category = category
        this.title = title
        this.cover = cover
        this.playlistUrl = playlistUrl
        this.originalId = originalId
    }

    companion object {
        // Static constants for license types
        const val LICENSE_TYPE_WIDEVINE: String = "com.widevine.alpha"
        const val LICENSE_TYPE_CLEAR_KEY: String = "clearkey"
        const val LICENSE_TYPE_CLEAR_KEY_2: String = "org.w3.clearkey"
        const val LICENSE_TYPE_PLAY_READY: String = "com.microsoft.playready"
    }
}
